clients/js/test/defaultGuards/assetBurn.test.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  assertBotTax,
+  assertBurnedAsset,
+  assertSuccessfulMint,
+  createAssetWithCollection,
+  createCollection,
+  createUmi,
+  createV2,
+  defaultAssetData,
+} from '../_setup';
+
+test('it burns a specific Asset to allow minting', async (t) => {
+  // Given the identity owns an Asset from a certain collection.
+  const umi = await createUmi();
+  const [assetToBurn, requiredCollection] = await createAssetWithCollection(
+    umi
+  );
+
+  // And a loaded Candy Machine with an assetBurn guard on that collection.
+  const collection = (await createCollection(umi)).publicKey;
+  const { publicKey: candyMachine } = await createV2(umi, {
+    collection,
+    configLines: [{ name: 'Degen #1', uri: 'https://example.com/degen/1' }],
+    guards: {
+      assetBurn: some({ requiredCollection: publicKey(requiredCollection) }),
+    },
+  });
+
+  // When the identity mints from it using its Asset to burn.
+  const mint = generateSigner(umi);
+  await transactionBuilder()
+    .add(setComputeUnitLimit(umi, { units: 600_000 }))
+    .add(
+      mintV1(umi, {
+        candyMachine,
+        asset: mint,
+        collection,
+        mintArgs: {
+          assetBurn: some({
+            tokenStandard: TokenStandard.NonFungible,
+            requiredCollection: publicKey(requiredCollection),
+            asset: assetToBurn.publicKey,
+          }),
+        },
+      })
+    )
+    .sendAndConfirm(umi);
+
+  // Then minting was successful.
+  await assertSuccessfulMint(t, umi, { mint, owner: umi.identity });
+
+  // And the Asset was burned.
+  await assertBurnedAsset(t, umi, assetToBurn);
+});
+
+test('it allows minting even when the payer is different from the minter', async (t) => {
+  // Given a separate minter owns an Asset from a certain collection.
+  const umi = await createUmi();
+  const minter = await generateSignerWithSol(umi);
+  const [, requiredCollection] = await createAssetWithCollection(umi);
+
+  const assetToBurn = generateSigner(umi);
+  await createV1(umi, {
+    ...defaultAssetData(),
+    asset: assetToBurn,
+    collection: requiredCollection.publicKey,
+    owner: minter.publicKey,
+  }).sendAndConfirm(umi);
+
+  // And a loaded Candy Machine with an assetBurn guard on that collection.
+  const collection = (await createCollection(umi)).publicKey;
+  const { publicKey: candyMachine } = await createV2(umi, {
+    collection,
+    configLines: [{ name: 'Degen #1', uri: 'https://example.com/degen/1' }],
+    guards: {
+      assetBurn: some({ requiredCollection: publicKey(requiredCollection) }),
+    },
+  });
+
+  // When the minter mints from it using its Asset to burn.
+  const mint = generateSigner(umi);
+  await transactionBuilder()
+    .add(setComputeUnitLimit(umi, { units: 600_000 }))
+    .add(
+      mintV1(umi, {
+        candyMachine,
+        asset: mint,
+        minter,
+        collection,
+        mintArgs: {
+          assetBurn: some({
+            tokenStandard: TokenStandard.NonFungible,
```

Comment:
Why do we still have the tokenStandard in here?
Wouldn't it make sense to remove it?